bvm/ballerina-profiler/src/main/java/io/ballerina/runtime/profiler/Profiler.java

```diff
@@ -0,0 +1,373 @@
+    }
+
+    private void handleProfilerArguments(String[] args) {
+        if (args.length == 0) {
+            return;
+        }
+        List<String> usedArgs = new ArrayList<>();
+        for (int i = 0; i < args.length; i++) {
+            switch (args[i]) {
+                case "--file" -> {
+                    this.balJarName = extractFileArg(args[i + 1]);
+                    addToUsedArgs(args, usedArgs, i);
+                }
+                case "--args" -> {
+                    this.balJarArgs = extractBalJarArgs(args[i + 1]);
+                    addToUsedArgs(args, usedArgs, i);
+                }
+                case "--skip" -> {
+                    this.skipFunctionString = extractSkipFunctionString(args[i + 1]);
+                    addToUsedArgs(args, usedArgs, i);
+                }
+                case "--target" -> {
+                    this.targetDir = args[i + 1];
+                    addToUsedArgs(args, usedArgs, i);
+                }
+                default -> handleUnrecognizedArgument(args[i], usedArgs);
+            }
+        }
+    }
+
+    private String extractFileArg(String value) {
+        if (!value.endsWith(".jar")) {
+            throw new ProfilerException("Invalid file argument found: " + value);
+        }
+        return value;
+    }
+
+    private String extractBalJarArgs(String value) {
+        if (value == null || !value.startsWith("[") || !value.endsWith("]")) {
+            throw new ProfilerException("Invalid JAR arguments found: " + value);
+        }
+        return value.substring(1, value.length() - 1);
+    }
+
+    private String extractSkipFunctionString(String value) {
+        if (value == null || !value.matches("\\[.*\\]")) {
+            throw new ProfilerException("Invalid skip functions found: " + value);
+        }
+        return value.substring(1, value.length() - 1);
+    }
+
+    private void handleUnrecognizedArgument(String argument, List<String> usedArgs) {
+        if (!usedArgs.contains(argument)) {
+            throw new ProfilerException("Unrecognized argument found: " + argument);
+        }
+    }
+
+    private void addToUsedArgs(String[] args, List<String> usedArgs, int i) {
+        usedArgs.add(args[i]);
+        usedArgs.add(args[i + 1]);
+    }
+
+    private void extractProfiler() throws ProfilerException {
+        OUT_STREAM.printf("%s[1/6] Initializing...%s%n", Constants.ANSI_CYAN, Constants.ANSI_RESET);
+        try {
+            new ProcessBuilder("jar", "xvf", "Profiler.jar", "io/ballerina/runtime/profiler/runtime").start().waitFor();
+        } catch (IOException | InterruptedException exception) {
+            throw new ProfilerException(exception);
+        }
+    }
+
+    private void createTempJar() {
+        try {
+            OUT_STREAM.printf("%s[2/6] Copying the executable...%s%n", Constants.ANSI_CYAN, Constants.ANSI_RESET);
```

Comment:
```suggestion
            OUT_STREAM.printf("%s[2/6] Copying executable...%s%n", Constants.ANSI_CYAN, Constants.ANSI_RESET);
```
When compiling the program we print `Generating executable` without `the`, hence I think here we can omit it too. Please excuse me for suggesting to changing this to former